packages/x-components/src/x-modules/tagging/store/actions/track-tagging.action.ts

```diff
@@ -0,0 +1,42 @@
+import { TaggingInfo } from '../../../../../../search-types';
+import { XPlugin } from '../../../../plugins/x-plugin';
+import { DefaultSessionService } from '../../service';
+import { TaggingXStoreModule } from '../types';
+
+/**
+ * Default implementation for the {@link TaggingActions.track}.
+ *
+ * @param context - The {@link https://vuex.vuejs.org/guide/actions.html | context} of the actions,
+ * provided by Vuex.
+ * @param taggingInfo - The taggingInfo.
+ *
+ * @public
+ */
+export const track: TaggingXStoreModule['actions']['track'] = (context, taggingInfo) => {
+  const consent = context.state.consent;
+  const tagging = Array.isArray(taggingInfo) ? taggingInfo : [taggingInfo];
+  const sessionId = getSessionId(consent);
+
+  tagging.forEach(({ url, params }: TaggingInfo) => {
+    return XPlugin.adapter.track({
+      url,
+      params: {
+        ...params,
+        session: sessionId
+      }
+    });
+  });
+};
+
+/**
+ * Returns the session id if the consent is true.
+ *
+ * @param consent - User consent to retrieve the session id.
+ *
+ * @returns SessionId - The user session id or undefined.
```

Comment:
You can remove `SessionId -` here, as it is not a param, just the returned value.

```suggestion
 * @returns The user session id or undefined.
```